app/Console/Commands/SetUploadUsers.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace App\Console\Commands;
+
+use App\Models\UploadUser;
+use Illuminate\Console\Command;
+use Illuminate\Support\Facades\Storage;
+
+class SetUploadUsers extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'uploadUsers:set {allowList}';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Set the list of users who are allowed to upload mismatch files by providing a txt file;';
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return int
+     */
+    public function handle()
+    {
+        $allowlistFile = 'allowlist/' . $this->argument('allowList');
+        echo "Trying to read allow list from storage/app/" . $allowlistFile . "\n";
+
+        if (!Storage::disk('local')->exists($allowlistFile)) {
+            echo "File not found\n";
+            return 1;
+        }
+
+        UploadUser::truncate();
+
+        // read file from local disc, split by newline and filter out empty lines
+        $lines = array_filter(
+            explode("\n", Storage::disk('local')->get($allowlistFile))
+        );
```

Comment:
This is very convenient, but array_filter will also filter out "0". To make this more reliable, I would suggest to not use it, but rather `trim( $line ) === ''` (for each line).

```
php > print_r(array_filter(["0"]));
Array
(
)
```

Reply:
tbh, I still prefer the convenience of array_filter over trimming each line one-by-one. Stackoverflow has an [elegant solution to the "0" problem](https://stackoverflow.com/a/38081606), but I would really find it rather strange to include the "strlen" callback. If the next reviewer asks what it is for, I'd have to add a comment to the code to explain that this will allow the username "0", which I am pretty sure will never show up on our allow list.